components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4715,4 +4719,28 @@ public static String resolveExternalConsentPageUrl(String tenantDomain) throws I
 
         return externalConsentPageUrl;
     }
+
+    /**
+     * Check whether the application should be FAPI conformant.
+     *
+     * @param clientId       Client ID of the application.
+     * @return Whether the application should be FAPI conformant.
+     * @throws OAuthClientAuthnException
+     */
+    public static boolean isFapiConformantApp(String clientId) throws OAuthClientAuthnException {
+
+        try {
+            ServiceProvider serviceProvider = getServiceProvider(clientId);
+            ServiceProviderProperty[] serviceProviderProperties = serviceProvider.getSpProperties();
+            for (ServiceProviderProperty serviceProviderProperty : serviceProviderProperties) {
+                if (IS_FAPI_CONFORMANT_APP.equals(serviceProviderProperty.getName())) {
+                    return Boolean.parseBoolean(serviceProviderProperty.getValue());
+                }
+            }
+        } catch (IdentityOAuth2Exception e) {
```

Comment:
Throw a server error instead of the client error. Also we can throw the IdentityOAuth2Exception itself.

Reply:
When calling this method we already have IdentityOauth2Exception covered
https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2151/files#diff-debebfb0b01b553afd77f73d24e1df466030d3d19b218e0d7b3005364a05939aR299

Reply:
Fixed in a935fe8cd8a4955e069bdab4cc3180223d1f2935